harness/features/variable.local.test.ts

```diff
@@ -0,0 +1,325 @@
+                await createUser(url, { name: 'invalid' })
+            ).headers.get('location')
+        })
+
+        describeIf(capabilities.includes(Capabilities.local))(name, () => {
+            describe('uninitialized client', () => {
+                it('should return default value if client is uninitialized',  async () => {
+                    const variableResponse = await callVariable(
+                        clientId, url, variationOnUser.location, 'string-var', 'string default'
+                    )
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+
+                it('should return default value if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( clientId, url, invalidUser.location, 'string-var', 'string default')
+                    const variable = await variableResponse.json()
+                    expectDefaultValue(variable, 'string default')
+                })
+            })
+
+            describe('initialized client', () => {
+                beforeAll(async () => {
+                    // mock endpoints
+                    scope
+                        .get(`/client/${clientId}/config/v1/server/dvc_server_test_token.json`)
+                        .reply(200, config)
+
+                    // wait until initialized
+                    await callOnClientInitialized(clientId, url, mockServerUrl, scope)
+                })
+
+                it('should throw exception if user is invalid',  async () => {    
+                    const variableResponse =
+                        await callVariable( clientId, url, invalidUser.location, 'string-var', 'string default')
+                    const variable = await variableResponse.json()
+
+                    expect(variable.exception).toBe('Must have a user_id set on the user')
+                })
+
+                forEachVariableType((type) => {
+                    const { key, defaultValue, variationOn } = expectedVariablesByType[type]
+
+                    it('should return variable if mock server returns object matching default type',  async () => {
+                        let eventBody = {}
+                        scope.post(`/client/${clientId}/v1/events/batch`).reply((uri, body) => {
+                            eventBody = body
+                            return [200]
+                        })
+                    
+                        const variableResponse = await callVariable(
+                            clientId, 
+                            url, 
+                            variationOnUser.location, 
+                            key, 
+                            defaultValue
+                        )
+                        const variable = await variableResponse.json()
+                        expect(variable.entityType).toBe('Variable')
```

Comment:
might be easier to read these assertions as just a single object matcher like 
```
expect(variable).toEqual({
  isDefaulted: false
  ...
})
```

this would also let you be explicit about allowing / not allowing additional properties

Reply:
you can add an `expect.matchObject` on that field or something

Reply:
yeah i just got the toEqual working and then it worked!